[nrfconnect] Added on-boarding codes generation to the factory data script (#25879)
project-chip/connectedhomeip: docs/guides/nrfconnect_factory_data_configuration.md

### Generating onboarding codes

The
[generate_nrfconnect_chip_factory_data.py](../../scripts/tools/nrfconnect/generate_nrfconnect_chip_factory_data.py)
script allows generating a manual code and QR code from given factory data
parameters. You can use those codes to perform commissioning to the Matter
network over Bluetooth LE since they include all the needed pairing data needed
by a Matter controller. You should place the codes on a device package or on the
device itself during a production step. To generate a manual pairing code and QR
code complete the following steps:

1. Install all needed python dependencies for Matter:

    ```
    $ python -m pip install -r ./scripts/setup/requirements.nrfconnect.txt
    ```

2. Complete steps `1, 2 and 3` from
   [Creating the factory data JSON file with the first script](#creating-the-factory-data-json-file-with-the-first-script)
   section to prepare a final invocation of the Python script.

3. Add the `--generate_onboarding` argument to the Python script final
   invocation:

    ```
    $ python scripts/tools/nrfconnect/generate_nrfconnect_chip_factory_data.py ...<arguments>... --generate_onboarding
    ```

4. Run the script.

> 5. Navigate to the output directory provided as `-o` argument.

Reviewer: ```suggestion
5. Navigate to the output directory provided as the `-o` argument.
```

> 6. Look for three files:
> 
>     - .json file containing the latest factory data set.
>     - .txt file containing the generated manual code and QRCode in a text
>       version.
>     - .png file containing the generated QRCode as an image.

Reviewer: ```suggestion
The output directory contains the following files you need:

- JSON file containing the latest factory data set.
- Test file containing the generated manual code and the text version of the QR Code.
- PNG file containing the generated QR Code as an image.
```

To generate a manual pairing code and a QR code, complete the following steps:

1. Install all required Python dependencies for Matter:

    ```
    $ python -m pip install -r ./scripts/setup/requirements.nrfconnect.txt
    ```

2. Complete steps 1, 2, and 3 from the
   [Creating the factory data JSON file with the first script](#creating-the-factory-data-json-file-with-the-first-script)
   section to prepare the final invocation of the Python script.

3. Add the `--generate_onboarding` argument to the Python script final
   invocation.

4. Run the script.

5. Navigate to the output directory provided as the `-o` argument.

The output directory contains the following files you need:

-   JSON file containing the latest factory data set.
-   Test file containing the generated manual code and the text version of the
    QR Code.
-   PNG file containing the generated QR Code as an image.

#### Enabling onboarding codes generation within the build system

You can generate onboarding codes using the nRF Connect platform build system
described in
[Building an example with factory data](#building-an-example-with-factory-data),
and build an example with the following additional option:

> `-DCONFIG_CHIP_FACTORY_DATA_GENERATE_ONBOARDING_CODES=y`:

Reviewer: ```suggestion
`-DCONFIG_CHIP_FACTORY_DATA_GENERATE_ONBOARDING_CODES=y`.
```
